src/main/java/org/example/StudentsGroup.java

```diff
@@ -0,0 +1,59 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class StudentsGroup {
+    private Student groupLeader;
+    private List<Student> studentsList;
+    private List<String> tasksList;
+    private Map<Student, List<Integer>> completedTasks;
```

Comment:
You are using Integer to mark task which compleated but what if we change tasksList and remove first task? Better to use this structure `Map<String, List<Student>>`